graphix/pauli.py

```diff
@@ -19,6 +21,58 @@ class IXYZ(enum.Enum):
     Z = 2
 
 
+class Sign(enum.Enum):
+    Plus = 1
+    Minus = -1
+
+    def __str__(self) -> str:
+        if self == Sign.Plus:
+            return "+"
+        return "-"
+
+    @staticmethod
+    def plus_if(b: bool) -> Sign:
+        if b:
+            return Sign.Plus
+        return Sign.Minus
+
+    @staticmethod
+    def minus_if(b: bool) -> Sign:
+        if b:
+            return Sign.Minus
+        return Sign.Plus
+
+    def __neg__(self) -> Sign:
+        return Sign.minus_if(self == Sign.Plus)
+
+    @typing.overload
+    def __mul__(self, other: Sign) -> Sign: ...
+
+    @typing.overload
+    def __mul__(self, other: Number) -> Number: ...
+
+    def __mul__(self, other):
```

Comment:
Please do not omit type annotations even for the base implementation.

Reply:
Are you suggesting that the base implementation should have the following signature?
```python
def __mul__(self, other: Sign | Number) -> Sign | Number:
```
While I understand the intent, I’m not sure this is the best approach. It introduces repetition in type annotations and can obscure the relationship between the input type and the return type. An alternative could be:
```
SignNumber = TypeVar("SignNumber", Sign | Number)
def __mul__(self, other: SignNumber) -> SignNumber:
```
However, I believe this might complicate maintenance rather than simplify it. According to [PEP 484](https://peps.python.org/pep-0484/), type annotations are often omitted in the actual implementation (for example, see the implementation of `def utf8`).

Reply:
> It introduces repetition in type annotations and can obscure the relationship between the input type and the return type.

I'm suggesting this fix to enable mypy check for the implementation body.
If you were to omit it, you would lose a chance to find hidden bugs, as `other` is now untyped.
I know this approach is not perfect and causes repetition, but still necessary to improve code quality.

Reply:
By the way, mypy is complaining that `Operator "*" not supported for types "Literal[1, -1]" and "Number"
`.